### ISSUE TITLE
Write output of  --file-list to file in $BuildDir instead of stdout

### ISSUE THREAD

**Issue by flxo** [esrlabs/bake#130]
The format of the `--file-list` output is quite nasty to handle because of two things:

* The `HEADER` and `FILE` prefixes are kind of a useless obstacle.
* The output is mixed with other `bake` output like this `Building ....` lines.

See a (stripped) example here:
```
...
HEADER: prebuilts/clang/linux-x86_64/arm-linux-androideabi-4.9/sysroot/usr/include/sys/types.h
HEADER: prebuilts/clang/linux-x86_64/arm-linux-androideabi-4.9/sysroot/usr/include/sys/ucontext.h
HEADER: prebuilts/clang/linux-x86_64/arm-linux-androideabi-4.9/sysroot/usr/include/sys/user.h
HEADER: prebuilts/clang/linux-x86_64/arm-linux-androideabi-4.9/sysroot/usr/include/unistd.h
**** Building 3 of 7: Mindroid.cpp (default) ****
FILE: src/mindroid/app/ContextImpl.cpp
FILE: src/mindroid/app/IOnSharedPreferenceChangeListener.cpp
FILE: src/mindroid/app/Service.cpp
FILE: src/mindroid/app/SharedPreferencesImpl.cpp
...
```

To make the feature more easy to use (without some grep etc processing) the output could be written to the `$BuildDir`. If the information about the type of a file is really needed the output could be formatted in `json` (like `--incs-and-defs`) to be more consistent.

**Comment by aschaal** [comment]
Ok, I will write a file, makes more sense.

**Comment by flxo** [comment]
Solved with v2.29.2
